test/multiverse/suites/unicorn/unicorn_test.rb

```diff
@@ -0,0 +1,41 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'unicorn'
+require 'newrelic_rpm'
+
+class UnicornTest < Minitest::Test
+  include MultiverseHelpers
+  attr_accessor :server
+
+  def setup
+    File.truncate('log/newrelic_agent.log', 0)
+    @server = ::Unicorn::HttpServer.new(
+      Rack::Builder.app(lambda { [200, {'Content-Type' => 'text/html'}, ['OK']] })
+    )
+    @server.start
+    trigger_agent_reconnect
+  end
+
+  def teardown
+    @server.stop
+  end
+
+  def test_unicorn_set_as_discovered_dispatcher
+    assert_logged('Dispatcher: unicorn')
+  end
+
+  def test_defer_message_logged_when_unicorn_in_use
+    assert_logged('Deferring startup of agent reporting thread because unicorn may fork.')
+  end
+
+  def assert_logged(expected_message)
+    messages = File.readlines('log/newrelic_agent.log')
```

Comment:
Does the startup of Unicorn happen in such a way that we can't hang an expectation on the logger itself, necessitating the reading of the log file instead?

Reply:
That's what I've noticed. The New Relic logger, though it is the object adding these items to the agent log, cannot find the reference to the printed unicorn log lines. In my tests, I may have been trying to access the logger incorrectly. I'd be interested in pairing on this if you'd like to take a look with me.

Reply:
Turns out, I was missing a stub in the setup block to allow me to connect with the logger object. This should be fixed now!